Rude/Rude.cs

```diff
@@ -8,208 +8,77 @@
-        }
 
-        private string GetUsername(int userId, string name)
+        private void EnsureRudeStatus(int userId)
         {
-            var inRudes = GetActiveInRudes(userId).Count();
-            var reudigLevel = GetReudigLevel(userId);
-            return GetUsername(name, inRudes, reudigLevel);
+            if (!RudeStatus.ContainsKey(userId))
+                RudeStatus[userId] = new RudeStatus
+                {
+                    OnRudeLevelDecrease = async () =>
+                    {
+                        var users = await Server.GetOnlineUsers();
+                        var user = users.SingleOrDefault(a => a.UserId == userId);
+                        if (user == null)
+                            return;
+
+                        user.Name = GetUserName(user);
+                        await user.SaveChanges();
+                    }
+                };
         }
 
-        private string GetUsername(string name, int inRudes, int reudigLevel)
+        private string GetUserName(IUser user)
         {
-            if (name.Contains("]"))
-                name = name.Substring(name.IndexOf("]") + 2);
-            // if actor gets punished, it is a completely valid rude on himself
-            //  allowing an increase of inRudes to a value over 3
-            var index = Math.Min(inRudes - 1, 2);
-            name = PREFIXES[index] + " " + name;
-
-            // a user can have a reudigLevel, but only be Ruderer if his first rude has timed out
-            if (inRudes >= PREFIXES.Length && reudigLevel > 0)
-                name.Insert(1, reudigLevel.ToString());
-            return name;
-        }
+            EnsureRudeStatus(user.UserId);
 
-        private IEnumerable<RudeEntity> GetActiveInRudes(int userId)
-        {
-            return Rudes.Where(r => r.TargetId == userId)
-                // take only rudes, which have not kicked the target
-                .Where(r => r.Duration != null)
-                // and which are still active
-                .Where(r => r.Timestamp + r.Duration >= DateTimeOffset.Now);
-        }
+            string name = user.Name;
 
-        private IEnumerable<RudeEntity> GetKickRudes(int userId)
-        {
-            var activeInRudes = GetActiveInRudes(userId);
-            return GetKickRudes(userId, activeInRudes);
-        }
-        private IEnumerable<RudeEntity> GetKickRudes(int userId, IEnumerable<RudeEntity> activeInRudes)
-        {
-            var lastActiveTimestamp = activeInRudes.Max(r => r.Timestamp);
-            return Rudes.Where(r => r.TargetId == userId)
-                // count rudes kicking him since last actve rude
-                .Where(r => r.Duration == null)
-                .Where(r => r.Timestamp > lastActiveTimestamp);
-        }
-        private int GetReudigLevel(int userId)
-        {
-            return GetKickRudes(userId).Count();
-        }
+            // Remove potential old prefix
```

Comment:
Yeah, but at the moment I don't want to keep track of applied prefixes. It's on my todo-list though.